xla/hlo/analysis/hlo_replication_analysis_test.cc

```diff
@@ -679,61 +679,73 @@ HloModule PartiallyReplicatedAllGatherFlattenedID
-          module.get(),
+          module_replica_analysis.get(),
           /*cross_partition_spmd=*/false));
+
+  TF_ASSERT_OK_AND_ASSIGN(
+      auto module_partition_analysis,
+      ParseAndReturnVerifiedModule(module_str, /*replica_count=*/2,
+                                   /*num_partitions=*/4));
   TF_ASSERT_OK_AND_ASSIGN(
       std::unique_ptr<HloReplicationAnalysis> partition_analysis,
       HloReplicationAnalysis::RunWithPartialReplication(
-          module.get(),
+          module_partition_analysis.get(),
           /*cross_partition_spmd=*/true));
 
   std::array<ReplicaGroup, 2> replica_groups0;
   replica_groups0[0].add_replica_ids(0);
-  replica_groups0[0].add_replica_ids(0);
-  replica_groups0[1].add_replica_ids(1);
-  replica_groups0[1].add_replica_ids(1);
+  replica_groups0[0].add_replica_ids(1);
+  replica_groups0[1].add_replica_ids(2);
+  replica_groups0[1].add_replica_ids(3);
 
   std::array<ReplicaGroup, 2> replica_groups1;
   replica_groups1[0].add_replica_ids(0);
-  replica_groups1[0].add_replica_ids(1);
-  replica_groups1[1].add_replica_ids(0);
+  replica_groups1[0].add_replica_ids(2);
   replica_groups1[1].add_replica_ids(1);
+  replica_groups1[1].add_replica_ids(3);
 
   EXPECT_FALSE(replica_analysis->HloInstructionIsReplicatedAt(
-      FindInstruction(module.get(), "all-gather0"), {}));
+      FindInstruction(module_replica_analysis.get(), "all-gather0"), {}));
 
   EXPECT_TRUE(replica_analysis->HloInstructionIsReplicatedAt(
-      FindInstruction(module.get(), "all-gather0"), {}, replica_groups0));
+      FindInstruction(module_replica_analysis.get(), "all-gather0"), {},
+      replica_groups0));
 
-  EXPECT_TRUE(replica_analysis->HloInstructionIsReplicatedAt(
-      FindInstruction(module.get(), "all-gather1"), {}));
+  EXPECT_FALSE(replica_analysis->HloInstructionIsReplicatedAt(
+      FindInstruction(module_replica_analysis.get(), "all-gather1"), {}));
 
   EXPECT_TRUE(replica_analysis->HloInstructionIsReplicatedAt(
-      FindInstruction(module.get(), "all-gather1"), {}, replica_groups1));
+      FindInstruction(module_replica_analysis.get(), "all-gather1"), {},
+      replica_groups1));
 
-  EXPECT_TRUE(partition_analysis->HloInstructionIsReplicatedAt(
-      FindInstruction(module.get(), "all-gather0"), {}));
+  EXPECT_FALSE(partition_analysis->HloInstructionIsReplicatedAt(
+      FindInstruction(module_partition_analysis.get(), "all-gather2"), {}));
 
   EXPECT_TRUE(partition_analysis->HloInstructionIsReplicatedAt(
-      FindInstruction(module.get(), "all-gather0"), {}, replica_groups1));
+      FindInstruction(module_partition_analysis.get(), "all-gather2"), {},
+      replica_groups0));
 
   EXPECT_FALSE(partition_analysis->HloInstructionIsReplicatedAt(
-      FindInstruction(module.get(), "all-gather1"), {}));
+      FindInstruction(module_partition_analysis.get(), "all-gather3"), {}));
 
   EXPECT_TRUE(partition_analysis->HloInstructionIsReplicatedAt(
```

Comment:
It's worth having some tests where different replicas have different partition_ids in replica_groups. E.g., doing the following

num_partitions = 5
num_replicas = 2
cross_partition_spmd = true
all gather replica_groups = {{0,1,2,5,6},{3,4,7,8,9}}
replica_groups passed to HloInstructionIsReplicatedAt = {{0,1},{2,3,4}}